src/processor.cpp

```diff
@@ -1162,21 +1251,39 @@ class _Processor
     }
 
     void selfdestruct()
-    {
+    {        
+        if (ctxt->in_static_mode)
+            throw Exception(ET::staticModeViolation, "Operation selfdestruct is not allowed in static mode");
+
         auto recipient = gs.get(pop_addr(ctxt->s));
         ctxt->acc.pay_to(recipient.acc, ctxt->acc.get_balance());
         tx.selfdestruct_list.push_back(ctxt->acc.get_address());
         stop();
     }
 
-    void create()
+    void create_impl()
     {
+        if (ctxt->in_static_mode)
+            throw Exception(ET::staticModeViolation, "Operation create is not allowed in static mode");
+
+        const auto op = get_op();
+        assert(op == CREATE || op == CREATE2);
+
         const auto contractValue = ctxt->s.pop();
         const auto offset = ctxt->s.pop64();
         const auto size = ctxt->s.pop64();
         auto initCode = copy_from_mem(offset, size);
 
-        const auto newAddress = generate_address(ctxt->acc.get_address(), ctxt->acc.get_nonce());
+        Address newAddress;
+        if (op == Opcode::CREATE)
+        {        
+            newAddress = generate_address(ctxt->acc.get_address(), ctxt->acc.get_nonce());
+        } 
+        else if (op == Opcode::CREATE2)
+        {
+            const auto salt = ctxt->s.pop();
+            newAddress = generate_address_for_create2(ctxt->acc.get_address(), salt, contractValue);
```

Comment:
in create2, contractValue should be initCode

Reply:
fixed